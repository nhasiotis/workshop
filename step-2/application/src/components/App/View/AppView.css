.App {
    text-align: flex-start;
    padding: 0.625em;
}

.App select {
    margin: 0.625em 0;
}
