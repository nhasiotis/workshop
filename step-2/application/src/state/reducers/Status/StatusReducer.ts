import { ActionState } from "@asml-ui/actions";
import { AnyAction } from "redux";

import { Status, StatusMessage, StatusType } from "../../../models/Status/Status";
import { Actions } from "../../actions/Actions";

export const statusReducer = (
    state: Status = { message: "", type: StatusType.SUCCESS, visible: false },
    action: AnyAction
): Status => {
    switch (action.type) {
        case Actions.FetchData.getType(ActionState.Succeed):
            return {
                message: StatusMessage.DATA_LOADED_SUCCESSFULLY,
                type: StatusType.SUCCESS,
                visible: true
            };
        case Actions.FetchData.getType(ActionState.Failed):
            return {
                message: StatusMessage.DATA_LOADED_FAILED,
                type: StatusType.ERROR,
                visible: true
            };
        default:
            return state;
    }
};
