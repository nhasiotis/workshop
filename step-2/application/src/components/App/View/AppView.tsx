import "./AppView.css";
import "antd/dist/antd.css";

import { Alert } from "antd";
import React from "react";

import { Status } from "../../../models/Status/Status";
import SelectView from "../../Select/View/SelectView";

export interface AppProps {
    scanners: string[];
    status: Status;
}
export const AppView = (props: AppProps) => (
    <div className="App">
        {props.status.visible ? (
            <Alert
                data-id="alert-msg-id"
                message={props.status.message}
                type={props.status.type}
            />
        ) : null}
        <SelectView items={props.scanners} id={"scanner-select"} />
    </div>
);
