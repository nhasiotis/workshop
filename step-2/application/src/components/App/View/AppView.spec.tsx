import { shallow } from "enzyme";
import React from "react";

import { StatusType } from "../../../models/Status/Status";
import { AppView } from "./AppView";

describe("Given AppView", () => {
    describe("when rendering", () => {
        it("should have AppView with alert component and list of scanners matching snapshot", () => {
            const expectedStatusVisible = {
                message: "Data is loaded successfully",
                type: StatusType.SUCCESS,
                visible: true
            };
            const expectedScanners = ["scannerA", "scannerB"];
            const appView = shallow(
                <AppView
                    scanners={expectedScanners}
                    status={expectedStatusVisible}
                />
            );

            expect(appView).toMatchSnapshot();
        });

        it("should have AppView without alert component and empty scanner list matching snapshot", () => {
            const expectedstatusNotVisible = {
                message: "Data is not loaded",
                type: StatusType.ERROR,
                visible: false
            };
            const appView = shallow(
                <AppView scanners={[]} status={expectedstatusNotVisible} />
            );

            expect(appView).toMatchSnapshot();
        });
    });
});
