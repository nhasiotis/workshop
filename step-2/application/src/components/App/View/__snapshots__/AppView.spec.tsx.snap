// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given AppView when rendering should have AppView with alert component and list of scanners matching snapshot 1`] = `
<div
  className="App"
>
  <Alert
    data-id="alert-msg-id"
    message="Data is loaded successfully"
    type="success"
  />
  <SelectView
    id="scanner-select"
    items={
      Array [
        "scannerA",
        "scannerB",
      ]
    }
  />
</div>
`;

exports[`Given AppView when rendering should have AppView without alert component and empty scanner list matching snapshot 1`] = `
<div
  className="App"
>
  <SelectView
    id="scanner-select"
    items={Array []}
  />
</div>
`;
